src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -603,6 +614,16 @@ private static LLVMValueRef CastIntValue(LLVMBuilderRef builder, LLVMValueRef va
             }
         }
 
+        private static LLVMValueRef CastDoubleValue(LLVMBuilderRef builder, LLVMValueRef value, LLVMTypeRef type)
+        {
+            if (LLVM.TypeOf(value).Pointer == type.Pointer)
+            {
+                return value;
+            }
+            Debug.Assert(LLVM.TypeOf(value).TypeKind == LLVMTypeKind.LLVMFloatTypeKind);
```

Comment:
Wouldn't we end up here if we load a double as well?

Reply:
I think that's caught by `if (LLVM.TypeOf(value).Pointer == type.Pointer)` isn't it?  I added a test to load `Double.MaxValue` which passes.

Reply:
Ah, yep. Thanks for adding the test as well.